src/speculators/models/dflash/metrics.py

```
@@ -46,13 +52,24 @@ def compute_metrics(
     pos_idx = torch.arange(seq_len, device=logits.device) % block_size
     pos_idx = pos_idx.unsqueeze(0)  # shape: [1, T]
 
+    elementwise_ce = None
+    if per_position_loss_weight == "dpace":
+        elementwise_ce = ce_loss(logits, targets)
+        decay_mult = dpace_loss_weight(
+            elementwise_ce, loss_mask, block_size, alpha=dpace_alpha
+        )
+        decay_fn = lambda pos_idx: decay_mult
```

Ruff (0.15.20)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `decay_fn` as a `def`

(E731)